src/controllers/user.ts

```diff
@@ -1,21 +1,54 @@
 import { Request, Response } from 'express'
 import { body, check, validationResult } from 'express-validator'
+import { User } from '../entity'
+import { logger } from '../logger'
 
-export const getAccount = (req: Request, res: Response): Response => {
-  if (req.user) {
-    return res.json(req.user.toJSON())
+export const getAccount = async (req: Request, res: Response): Promise<Response> => {
+  const user = req.user
+
+  if (!user) {
+    return res.status(401).json({ error: 'not logged in.' })
   }
-  return res.status(501).json({ error: 'not implemented' })
+
+  return res.status(200).json({ user })
 }
 
-export const postUpdateProfile = async (req: Request, res: Response): Promise<Response> => {
-  await check('email', 'Please enter a valid email address.').isEmail().run(req)
-  await body('email').normalizeEmail({ gmail_remove_dots: false }).run(req)
+export const putUpdateProfile = async (req: Request, res: Response): Promise<Response> => {
+  if (!req.user) {
+    return res.status(401).json({ error: 'not logged in.' })
+  }
+
+  const email = req.body?.email
+  const name = req.body?.name
+
+  if (!email && !name) {
+    return res.status(422).json({ error: 'Name and/or Email are required.' })
+  }
+
+  if (email) {
+    await check('email', 'Please enter a valid email address.').isEmail().run(req)
+    await body('email').normalizeEmail({ gmail_remove_dots: false }).run(req)
+  }
+
+  if (name) {
+    await check('name', 'Name cannot be empty.').notEmpty().run(req)
+    await check('name', 'Name must be a string.').isString().run(req)
+  }
 
   const errors = validationResult(req)
 
   if (!errors.isEmpty()) {
     return res.status(422).json({ errors: errors.array() })
   }
-  return res.status(501).json({ error: 'no updates applied' })
+
+  try {
+    let user = req.user
+    user.email = email || user.email
+    user.name = name || user.name
+    user = await User.save(user)
+    return res.status(200).json({ user })
```

Comment:
I'm going to update this to use `toJSON` to be safe and to also allow all keys to be returned:
```suggestion
    return res.json({ user: user.toJSON() })
```
Also we can remove status `200` since it is inferred, only need to call `status` when it's anything but `200`.